### PR TITLE
chore(checkbox): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by nuria1110**
### Proposed behaviour

Convert tests to RTL.

### Current behaviour

Tests are in enzyme.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/checkbox/checkbox-group/checkbox-group.test.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  );
+
+  expect(screen.getByText("legend")).toHaveStyleRule(
+    "content",
+    '"(optional)"',
+    { modifier: "::after" }
+  );
+});
+
+test("should render error message and no tooltip when error prop is set and validationRedesignOptIn is true", () => {
```
Comment (edleeks87) [inline src/components/checkbox/checkbox-group/checkbox-group.test.tsx:69]: comment: I think it's overkill to test if the tooltip is there for this tbh

File: src/components/checkbox/checkbox-group/checkbox-group.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        data-role="checkbox-1"
+        value="1"
+        label="label-1"
+        onChange={() => {}}
+      />
+      <Checkbox value="2" label="label-2" onChange={() => {}} />
+    </CheckboxGroup>
+  );
+
+  // get styles set by :first-child
```
Comment (edleeks87) [inline src/components/checkbox/checkbox-group/checkbox-group.test.tsx:108]: question: can we not use `toHaveStyle` here as it asserts against the computed styles?

Comment (nuria1110) [reply src/components/checkbox/checkbox-group/checkbox-group.test.tsx:108]: You are absolutely right and I have no idea why I made it harder for myself by doing this

**Review by edleeks87 — DISMISSED** [review]

File: src/components/checkbox/checkbox.style.ts [hunk 3]
```diff
@@ -137,9 +137,6 @@ const StyledCheckbox = styled.div<StyledCheckboxProps>`
 
       ${StyledFieldHelp} {
         margin-left: 24px;
-        padding-left: ${labelSpacing === 1
```
Comment (nineteen88) [inline src/components/checkbox/checkbox.style.ts:140]: question: why has this been removed?

I take it it's no longer needed it just stood out with it being a RTL conversion that we're making a styling change here

Comment (nuria1110) [reply src/components/checkbox/checkbox.style.ts:140]: I removed this as I noticed it was already setting the labelSpacing padding in line [133](https://github.com/Sage/carbon/blob/42323ee6b1677d9317c7dcaa485eb9b53a0f93f8/src/components/checkbox/checkbox.style.ts#L113) and this was just copied over into the large size styles so it made no difference to remove this. For coverage we would have also needed another test which I thought was pointless.

Comment (nineteen88) [reply src/components/checkbox/checkbox.style.ts:140]: Okay, if it was just style bloat happy to remove it. I've just moved the PR to `ready for review` so we can just make sure Chromatic has no issues

**Review by nineteen88 — APPROVED** [review]

**Comment by edleeks87** [comment]
I approve ✅

**Pull request merged**